sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -181,10 +190,11 @@ __ESIMD_API std::enable_if_t<std::is_integral<T0>::value &&
                                  std::is_integral<U>::value,
                              __ESIMD_NS::simd<T0, SZ>>
 rol(__ESIMD_NS::simd<T1, SZ> src0, U src1) {
-  using ComputationTy = __ESIMD_DNS::computation_type_t<decltype(src0), U>;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src0 = src0;
-  typename __ESIMD_DNS::simd_type<ComputationTy>::type Src1 = src1;
-  return __esimd_rol<T0>(Src0.data(), Src1.data());
+  using ComputationTy =
+      __ESIMD_DNS::computation_type_t<decltype(src0), int32_t>;
```

Comment:
I doubt that the logic used for SHL and SHR should be used for ROL/ROR.

simd<int8_t, 2> in = {0x40, 0x7F};
simd<int8_t, 2> res = esimd::rol<int8_t>(in, 2); // roll left by 2 bits. Expected result: {0x01, 0xFD}

with the current code the result would be incorrect: {0x00, 0xFC}

Can you please add such-or-similar test case to LIT test?